### PR TITLE
Fix Parameter Resolution (both schema) & documentation for the triggerrule use case

### PR DESCRIPTION
**Pull request by mrutkows**

### CODE REVIEW

**Comment by mrutkows** [comment]
THis PR may also fix
 https://github.com/apache/incubator-openwhisk-wskdeploy/issues/297

**Comment by mrutkows** [comment]
@houshengbo @pritidesai @daisy-ycguo please review and merge.  Also, @priti can you please take the 2 manifests I created here (i.e., tests/dat/manifest_validate_multiline_params.yaml and tests/dat/manifest_validate_singleline_params.yaml) and turn these into unit test cases for the manifest_parser.go file (for unit testing the ResolveParameter() function)?

**Comment by pritidesai** [comment]
thanks @mrutkows yup have been working on creating those unit tests, will add them here in this PR.

**Comment by pritidesai** [comment]
Matt, I have tested your changes and have uploaded test cases here (marked it as WIP as adding more test cases):

https://github.com/apache/incubator-openwhisk-wskdeploy/pull/308

File: parsers/manifest_parser.go [hunk 1]
```diff
@@ -403,19 +425,151 @@ func (action *Action) ComposeWskAction(manipath string) (*whisk.Action, error) {
...
+		// we have a multi-line parameter declaration
+
+		// if we do not have a value, but have a default, use it for the value
+                if param.Value == nil && param.Default !=nil {
+			param.Value = param.Default
+		}
+
+		// if we also have a type at this point, verify value (and/or default) matches type, if not error
+		// Note: if either the value or default is in conflict with the type then this is an error
+		tempType, errorParser = ResolveParamTypeFromValue(param.Value)
```
Comment (pritidesai) [inline parsers/manifest_parser.go]: Here the comment says error should be thrown if we have a mismatch of specified "type" and "type" of the value. Try running it with an example of Parameter{Value: 11, Type: string}. ResolveParamTypeFromValue(11) determines the normalized type name which is "integer" and sets tempType to "integer". There is no check on actualType and type of the value done. This is how struct Parameter looks like:

```
(*parsers.Parameter)(0xc420163570)({
 Type: (string) (len=6) "string",
 Description: (string) "",
 Value: (int) 11,
 Required: (bool) false,
 Default: (interface {}) <nil>,
 Status: (string) "",
 Schema: (interface {}) <nil>,
 multiline: (bool) true
})
```  

Also, here the execution continues as there is no check on errorParser and return if errorParser is not `<nil>`

Comment (mrutkows) [inline parsers/manifest_parser.go]: Can we do this on a separate issue?  I would like to get this merged for the tests included thus far.  If we try to address all the TODOs I added regarding error/warning handling this issue will get too large IMO. @pritidesai

Comment (pritidesai) [inline parsers/manifest_parser.go]: sure, sounds good to me, thanks

**Pull request merged**